### PR TITLE
feat(BA-5419): support percentage values for rolling update `max_surge` and `max_unavailable`

### PR DESCRIPTION
**Pull request by jopemachine**
Resolves BA-5419.

## Summary
- Add support for float fraction values (e.g., `0.25` for 25%) for `max_surge` and `max_unavailable` rolling update parameters, in addition to existing absolute integer values
- Float values are resolved to absolute counts at execution time based on desired replica count, with `max_surge` rounding up and `max_unavailable` rounding down (matching Kubernetes semantics)
- Validation ensures float fractions are between 0.0 and 1.0, and existing integer-only configurations continue to work without changes

## Rounding semantics (follows Kubernetes convention)

| Field | Rounding | Reason |
|---|---|---|
| `max_surge` | **ceil** (rounds up) | Determines how many extra replicas can be created — rounding up ensures at least 1 can be created for progress |
| `max_unavailable` | **floor** (rounds down) | Determines how many replicas can be taken down — rounding down preserves availability conservatively |

**Example** (`desired_replicas = 10`, both set to `25%`):
- `max_surge`: `ceil(10 * 0.25) = ceil(2.5) = 3` → up to 13 replicas simultaneously
- `max_unavailable`: `floor(10 * 0.25) = floor(2.5) = 2` → at least 8 replicas always available

Both directions round toward **safety**: surge allows more creation headroom, unavailable restricts how many can go down.

## Test plan
- [x] DTO validation tests: float fraction input accepted, boundary values, negative rejected
- [x] RollingUpdateSpec tests: fraction resolve with rounding, passthrough for integers, deadlock rejection for 0+0
- [x] Strategy evaluation tests: correct route creation/termination with fractional specs
- [x] GQL mutation tests pass unchanged
- [ ] CI passes lint/check/test

Resolves BA-5419

----
📚 Documentation preview 📚: https://sorna--10563.org.readthedocs.build/en/10563/

----
📚 Documentation preview 📚: https://sorna-ko--10563.org.readthedocs.build/ko/10563/

### CODE REVIEW

File: src/ai/backend/manager/api/gql/deployment/types/policy.py [hunk 1]
```diff
@@ -28,6 +28,10 @@
     DeploymentStrategySpecInfo,
     RollingUpdateStrategySpecInfo,
 )
+from ai.backend.common.dto.manager.v2.deployment.types import (
+    IntOrPercent as IntOrPercentDTO,
```
Comment (seedspirit) [inline src/ai/backend/manager/api/gql/deployment/types/policy.py:32]: Is this an implementation that replaces the `@oneOf` directive?

File: src/ai/backend/manager/models/deployment_policy/row.py [hunk 2]
```diff
@@ -34,26 +36,52 @@
...
+    """Specification for rolling update deployment strategy.
 
-    max_surge: int = Field(default=1, ge=0)
-    max_unavailable: int = Field(default=0, ge=0)
+    ``max_surge`` and ``max_unavailable`` are :class:`IntOrPercent` objects.
+    Percentage values are resolved to absolute counts at execution time via
+    :meth:`resolve_max_surge` / :meth:`resolve_max_unavailable`.
+    """
+
+    max_surge: IntOrPercent = Field(default_factory=lambda: IntOrPercent(percent=0.5))
```
Comment (seedspirit) [inline src/ai/backend/manager/models/deployment_policy/row.py:46]: I think these default settings should be listed in the API documentation—are they mentioned there?

Comment (jopemachine) [reply src/ai/backend/manager/models/deployment_policy/row.py:46]: I added the description
